internal/pkg/deploy/cloudformation/stack/app.go

```diff
@@ -27,10 +27,23 @@ type DeployedAppMetadata struct {
 // AppResourcesConfig is a configuration for a deployed Application
 // StackSet.
 type AppResourcesConfig struct {
-	Accounts []string `yaml:"Accounts,flow"`
-	Services []string `yaml:"Services,flow"`
-	App      string   `yaml:"App"`
-	Version  int      `yaml:"Version"`
+	Accounts []string              `yaml:"Accounts,flow"`
+	Services []AppResourcesService `yaml:"Services,flow"`
+	App      string                `yaml:"App"`
+	Version  int                   `yaml:"Version"`
+}
+
+// AppResourcesService is a configuration for service for a deployed Application StackSet
+type AppResourcesService struct {
+	Name    string `yaml:"Name,flow"`
+	WithECR bool   `yaml:"WithECR,flow"`
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the Image
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v3) interface.
+func (s *AppResourcesService) UnmarshalYAML(value *yaml.Node) error {
+	return value.Decode(&s.Name)
```

Comment:
i'm confused how this will work when a _second_ service is created. i see how the flow works if you disable ecr for the first service, but what about when a new app template is generated for a _future_ `svc init`? won't the `WithECR` field always be false? so all of the ECR repos will be deleted?